ecs-cli/modules/utils/compose/convert_task_definition.go

```diff
@@ -263,6 +263,14 @@ func convertToContainerDef(context *project.Context, inputCfg *config.ServiceCon
 		outputContDef.WorkingDirectory = aws.String(inputCfg.WorkingDir)
 	}
 
+	outputContDef.LinuxParameters = &ecs.LinuxParameters{Capabilities: &ecs.KernelCapabilities{}}
+	if inputCfg.CapAdd != nil {
```

Comment:
is it possible that this will be empty instead of nil?

Reply:
Nope. Here is the line in libcompose for this field: 
```CapAdd          []string             yaml:"cap_add,omitempty"```

Which means that it will be nil if its not present. If on the other hand you specify it but don't give any capabilities (empty list), then you get this error:
```
ERRO[0000] Could not parse config for project test2 : Service 'mysql' configuration key 'cap_add' contains an invalid type, it should be an array.
```